main.go

```diff
@@ -89,40 +92,109 @@ func isDirEmpty(dirPth string) (bool, error) {
 }
 
 func isCacheAvailable(srcDir string, swiftVersion string) (bool, error) {
+	log.Printf("Check cache specific files")
+	buildDirAvailable, cacheFileAvailable, resolvedFileAvailable := true, true, true
+
 	// check for built dependencies (Carthage/Build/*)
+	fmt.Print("- Carthage/Build director: ")
 	carthageDir := filepath.Join(srcDir, carthageDirName)
 	carthageBuildDir := filepath.Join(carthageDir, buildDirName)
 	if empty, err := isDirEmpty(carthageBuildDir); err != nil {
```

Comment:
ouch `isDirEmpty` has some issue:

```
func isDirEmpty(dirPth string) (bool, error) {
	if exist, err := pathutil.IsDirExists(dirPth); err != nil {
		return false, err
	} else if !exist {
		return false, nil
	}
...
```

it returns `true` if the dir does not exist or if failed to check. So if you remove the `Carthage/Builds` dir it will report that it is not empty.

Reply:
I would rather go with a lot more simpler solution, for example:
```
files, err := ioutil.ReadDir(folderpath)
if err != nil {
    return false, err
}
return len(files)==0, nil
```

It is not worth to check the existence of the dir separately, as it will return the error anyway

Reply:
Fixed in:
https://github.com/bitrise-steplib/steps-carthage/pull/54/commits/0f6a949659ead23f09fa6055f80ec65c1ff776bd